### DISCUSSION TITLE
Does Slint for Rust use `winit` on Linux/X11?

### DISCUSSION THREAD

**Discussion by adrian5** [slint-ui/slint#2843] [Q&A]
Do understand it right that Slint uses [winit](https://github.com/rust-windowing/winit) on Linux/X11 targets?

I built the "gallery" example and `winit` was part of the compiled dependencies. I'm trying to narrow down the cause of a Rust GUI issue (excessive window redraws) that **Slint does not exhibit** but another frameworks does.

**Comment by tronical** [comment]
Yes, your analysis is correct. If you have Qt installed, that might be used instead. You can select a backed (winit or qt) at compile time if you disable default features and select either backend-winit or backend-qt. Or you can set the SLINT_BACKEND environment variable to switch before starting the process.

**Comment by ogoffart** [comment]
Slint has different backend.  By default it has a Qt and a winit backend are enabled.
The Qt backend result in a dummy backend if Qt is not found during compile time, but is otherwise the default (to enable the native style)
The winit backend is always compiled in by default (if it was possible to tell cargo not to compile it when Qt is found, we"d do that),
but it is not selected at runtime if Qt was there.

To select the backend, you can run with the environment variable

```
SLINT_BACKEND=winit
```

**Comment by adrian5** [comment]
Thank you both, very helpful!

Using the `SLINT_BACKEND` environment variable, I was able to test the two backends. The `winit` backend doesn't show the redraw issues which is valuable to know. Nonetheless, the Qt backend behaved considerably better for me — I think `winit` has a lot of work left to do.
